bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:assert", "assert")
+#@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:ip", "ip")
+
+
+#@ flexibleServerName    = data.values.resource_name
+#@ dbSecretName          = data.values.resource_name
+#@ resourceGroupK8SName  = data.values.resource_name
+#@ serviceAccountName      = data.values.resource_name + "-reader"
+#@ serviceAccountNamespace = data.values.resource_namespace
+#@ roleName                = data.values.resource_name + "-reading"
+
+#@ mandatory_fields = [ "location", "resource_group", "databases", "firewall_rules" ]
+#@ missing_fields = []
+#@ for field in mandatory_fields:
+#@   if len(data.values[field]) == 0:
+#@     missing_fields.append(field)
+#@   end
+#@ end
+#@ if len(missing_fields) > 0:
+#@   assert.fail("Missing values for mandatory fields '{}'".format(missing_fields))
+#@ end
+
+#@ if not regexp.match("^Standard_", data.values.instance_type):
```

Comment:
nit/question: I am not sure about the checks/assertions on `instance_type`, `instance_tier` & `instance_storage_size_gb`; after all we don't own the underlaying APIs ...

Reply:
These are required fields and have strict value requirements in the underlying (Azure RM) and the ASO v2 APIs. When we can, fail fast and aid the users.